### PR TITLE
Add screenshot service, devcontainer.

### PR DESCRIPTION
**Pull request by mattcburns**
## Description

The primary feature of this PR is to add screenshot support to the service that was earlier added to bmclib. In addition, I have added devcontainer support to make development easier on some platforms.

I have removed go-grpc validations from the protobuf definitions and grpc build process. This is due to incompatibility with building grpc libraries for languages other than Go. If there is some workaround (or I missed something), I am open to suggestions but otherwise it should be removed so other languages can use the client generation capabilities.

## How Has This Been Tested?
I have added a command line option under a new command subcategory called `diagnostic screenshot` to test capture of the screenshots via CLI. A user needs to set the `BMCIP` and `BMCPASS` environment variables and the command line tool will use these to connect to the target machine.

## How are existing users impacted? What migration steps/scripts do we need?

No major impact to end users - if they want to take advantage of the new options they will need to update any tooling around this application.

## Checklist:

I have:

- [X] updated the documentation and/or roadmap (if required)
- [X] added unit or e2e tests
- [X] provided instructions on how to upgrade

### CODE REVIEW

**Comment by jacobweinstock** [comment]
Hey @mattcburns, thanks for the PR! I'm not convinced that adding this amount of Ruby code is a good idea for the repo. The existing Ruby code was only there as a very basic example of a client. This amount of code feels like it's a library and now something we'll need to maintain. I'm not familiar enough with Ruby to want to take this on. I don't believe the other maintainers are familiar with Ruby either but i'll let them chime in. cc @joelrebel , @chrisdoherty4

**Comment by mattcburns** [comment]
> Hey @mattcburns, thanks for the PR! I'm not convinced that adding this amount of Ruby code is a good idea for the repo. The existing Ruby code was only there as a very basic example of a client. This amount of code feels like it's a library and now something we'll need to maintain. I'm not familiar enough with Ruby to want to take this on. I don't believe the other maintainers are familiar with Ruby either but i'll let them chime in. cc @joelrebel , @chrisdoherty4

I can just remove it, it's auto generated from the protobufs so we don't really need to keep it in there. Thanks for the feedback!

File: api/v1/bmc.proto
Comment (jacobweinstock) [inline api/v1/bmc.proto:22]: I'm not opposed to dropping these validations, I would just like to be sure that the validations are handled. I could be missing it, but i don't see them handled in any other way in the PR?

**Review by joelrebel — COMMENTED** [review]
Nice work, left a few suggestions inline.

File: api/v1/bmc.proto
Comment (joelrebel) [reply api/v1/bmc.proto:22]: I see Jacob got to this before I did, will set this to resolved.

File: cmd/screenshot.go
Comment (jacobweinstock) [reply cmd/screenshot.go:40]: FYI, CLI flags are already available as env vars through viper as `PBNJ_`. reference: https://github.com/tinkerbell/pbnj/blob/629921c5f6e4282ef1a2281a8995e5801b0ddeaa/cmd/root.go#L62C1-L64C1

If you define cobra cli flags the envs will come along with the viper integration. here's the `server` command example: https://github.com/tinkerbell/pbnj/blob/629921c5f6e4282ef1a2281a8995e5801b0ddeaa/cmd/server.go#L109

File: cmd/diagnostic.go
Comment (jacobweinstock) [inline cmd/diagnostic.go:8]: I'm not opposed to this but i don't understand the reason for this subcommand. `pbnj diagnostic screenshot` vs `pbnj screenshot`, right? curious what you're thinking/optimizing for here?

Comment (jacobweinstock) [reply cmd/diagnostic.go:8]: oh maybe because the protobuf service is called `diagnostic`?

File: grpc/oob/diagnostic/screenshot.go
Comment (jacobweinstock) [inline grpc/oob/diagnostic/screenshot.go:36]: This feels like it should be required always, no? if that is the case, you might consider making it explicit as a dependency in the `GetScreenshot` function.

File: api/v1/bmc.proto
Comment (mattcburns) [reply api/v1/bmc.proto:22]: I'm not sure how to support validations in the proto definitions without the library but with the library I can't generate client code for languages other than go. I'll do some more research to see what can be done for validations.
Do you happen to know what the original intent was behind adding the validations into the protobuf definitions?

File: cmd/screenshot.go
Comment (mattcburns) [reply cmd/screenshot.go:40]: Ah good point, I'll add that to the cobra config then

File: cmd/diagnostic.go
Comment (mattcburns) [reply cmd/diagnostic.go:8]: The idea is that in the future there may be other diagnostic commands other than screenshot - but it may be too early to start doing that; I can simplify it down to just screenshot.

File: api/v1/bmc.proto
Comment (mattcburns) [reply api/v1/bmc.proto:22]: Well, I went back to this again on the ruby side and found the right way to build this while keeping the validations, I'll update the PR shortly to reflect this.

**Comment by mattcburns** [comment]
I addressed all the feedback - I didn't realize that a force push after I rebased would remove most of those inline comments, so I apologize for that, but I think they're all still part of the conversation view. Thank you!

File: grpc/oob/diagnostic/screenshot.go
Comment (joelrebel) [inline grpc/oob/diagnostic/screenshot.go:80]: Theres a helper method here to retrieve the timeout set through env vars,

https://github.com/tinkerbell/pbnj/blob/629921c5f6e4282ef1a2281a8995e5801b0ddeaa/grpc/oob/common.go#L118

**Comment by mattcburns** [comment]
Addressed the additional feedback - this should be ready for further review @jacobweinstock  @joelrebel 
Thanks!

**Review by jacobweinstock — APPROVED** [review]
Thanks for this @mattcburns !

**Review by joelrebel — APPROVED** [review]

**Pull request merged**
